.github/workflows/autobuild.yml

```
@@ -14,178 +15,168 @@
 #     for every tag that starts with 'r' and does not have any suffix:
 #          - do CodeQl while building for every platform
 #          - publish the created binaries/packs only as artifacts/appendix as a release
-#
-
 
 on:
   workflow_dispatch:
   push:
     tags:
       - "r*"
     branches:
-      - "autobuild*"  # for developers: branches starting with autobuild will be built and evaluated on each push
-      - "autobuild/*" # for developers: branches starting with autobuild will be built and evaluated on each push
+      # For developers: Branches starting with autobuild will be built and evaluated on each push.
+      - "autobuild**"
       - "master"
-    #  - "*"
-  pull_request: # The branches below must be a subset of the branches in "push"
+  pull_request:
     branches:
+      # These branches must be a subset of the branches in "push":
       - master
 
 name:                               Auto-Build
 jobs:
   create_release:
-     name:                          Prepare Auto-Build/Release
-     runs-on:                       ubuntu-20.04
-     outputs:
+    # Check if we are doing a release or just a normal build.
+    # This must be done before actually building the app to find out where to upload the binaries and if we need to create a Github release.
+    name:                           Create Github release if this is a release build
```

> Looking at the workflow, I think the new name is slightly misleading. This step happens first, before the builds, so it isn't actually creating the Github release, just setting up variables to tell later steps to generate one. So "Prepare" or "Setup" is better, but happy for the rest of the name to be made clearer.

@softins By *new name* you are referring to the line which this comment is attached to, right?

This job does indeed create the Github release (although it doesn't fill it). The step some lines below is also named like that.

Maybe this would be better:
```suggestion
    name:                           Create empty Github release if this is a release build
```

---

Addition: I just realized that the changelog is generated by `anaylze_git_references.py`. Generating a file in an "analyze_" script certainly violates the principle of least surprise...

As this is both related to this PR and #2471, I think I'll open a separate Issue to track this and work on it when both PRs are merged. I've adjust the name of the `analyse_git_reference.py` step for now.

---

> > Looking at the workflow, I think the new name is slightly misleading. This step happens first, before the builds, so it isn't actually creating the Github release, just setting up variables to tell later steps to generate one. So "Prepare" or "Setup" is better, but happy for the rest of the name to be made clearer.
> 
> @softins By _new name_ you are referring to the line which this comment is attached to, right?
> 
> This job does indeed create the Github release (although without the artifacts, of course). The step some lines below is also named like that.
> 
> Example: https://github.com/jamulussoftware/jamulus/runs/5265414702?check_suite_focus=true#step:5:1

Ah, ok, I misunderstood. I thought it was just preparing variables for later use. So it's actually creating a release container that will later receive the assets.

My only other comment there would be that the `name` is getting rather long, and becoming more like a comment or description. Can it be made more concise but still accurate?

---

> My only other comment there would be that the name is getting rather long, and becoming more like a comment or description. Can it be made more concise but still accurate?

Hrm, you are right. I struggle to find good wording though. Maybe we should drop the imperative for job-level names.

If just adjusted it like that:
```suggestion
    name:                           Build vars & Github release (if required)
```